app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsPresenter.kt

```diff
@@ -268,10 +270,13 @@ class DepictsPresenter @Inject constructor(
      */
     fun getRecentDepictedItems(): MutableList<DepictedItem> {
         val depictedItemList: MutableList<DepictedItem> = ArrayList()
-        val depictsList = depictsDao.depictsList()
-        for (i in depictsList.indices) {
-            val depictedItem = depictsList[i].item
-            depictedItemList.add(depictedItem)
+        CoroutineScope(Dispatchers.IO).launch {
+            val depictsList = depictsDao.depictsList().await()
+
+            for (i in depictsList.indices) {
+                val depictedItem = depictsList[i].item
+                depictedItemList.add(depictedItem)
+            }
```

Comment:
This change may have triggered this bug: #5791
Any idea? :-)

Reply:
You got it right. The issue's here with the coroutine that I created. Because of its non-blocking nature the function is returned without waiting for the coroutine to complete.